file.go

```diff
@@ -175,10 +175,85 @@ func (f *File) setBatchesFromJSON(bs []byte) error {
 		if batches.Batches[i] == nil {
 			continue
 		}
-		if err := batches.Batches[i].build(); err != nil {
-			return batches.Batches[i].Error("Invalid Batch", err, batches.Batches[i].Header.ID)
+		batch := *batches.Batches[i]
+		batch.Header.recordType = batchHeaderPos
+
+		for _, e := range batch.Entries {
+			// these values need to be inferred from the json field names
+			e.recordType = "6"
+			if e.Addenda02 != nil {
+				e.Addenda02.recordType = "7"
+				e.Addenda02.TypeCode = "02"
+			}
+			for _, a := range e.Addenda05 {
+				a.recordType = "7"
+				a.TypeCode = "05"
+			}
+			if e.Addenda98 != nil {
+				e.Addenda98.recordType = "7"
+				e.Addenda98.TypeCode = "98"
+			}
+			if e.Addenda99 != nil {
+				e.Addenda99.recordType = "7"
+				e.Addenda99.TypeCode = "99"
+			}
+		}
+
+		if err := batch.build(); err != nil {
+			return batch.Error("Invalid Batch", err, batch.Header.ID)
+		}
+
+		// Attach a batch with the correct type
+		switch batch.Header.StandardEntryClassCode {
+		case ACK:
+			f.Batches = append(f.Batches, &BatchACK{batch})
```

Comment:
Every case here needs to get tested to pass code coverage. So a test JSON file for each batch type will need to be tested. That's why I just removed this file from code coverage.

Reply:
Creating a JSON file for each batch type would seem to be a separate issue(s).